spring-integration-amqp/src/test/java/org/springframework/integration/amqp/support/AmqpDedicatedChannelAdviceIntegrationTests.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.amqp.support;
+
+import static org.junit.Assert.assertTrue;
+
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.junit.RabbitAvailable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.amqp.dsl.Amqp;
+import org.springframework.integration.annotation.IntegrationComponentScan;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+/**
+ * @author Gary Russell
+ * @since 5.1
+ *
+ */
+@SpringJUnitConfig
+@RabbitAvailable(queues = AmqpDedicatedChannelAdviceIntegrationTests.QUEUE)
+public class AmqpDedicatedChannelAdviceIntegrationTests {
+
+	public static final String QUEUE = "dedicated.advice";
+
+	@Autowired
+	private Config.Gate gate;
+
+	@Autowired
+	private Config config;
+
+	@Test
+	public void testAdvice() throws Exception {
+		this.gate.send("a,b,c");
+		assertTrue(this.config.latch.await(10, TimeUnit.SECONDS));
```

Comment:
This test doesn't bring too much value and doesn't confirm how the new advice is cool.
I'm not sure what and how we should test here, but the result I'm sure I can get even without this advice.

Maybe really to check an order of messages with a single-threaded consumer?